ocaml-lsp-server/src/rename.ml

```diff
@@ -10,7 +10,9 @@ let rename (state : State.t)
     let command =
       Query_protocol.Occurrences (`Ident_at (Position.logical position), `Buffer)
     in
-    let+ locs = Document.Merlin.dispatch_exn ~name:"rename" merlin command in
+    let+ locs, _desync =
+      Document.Merlin.dispatch_exn ~name:"rename" merlin command
```

Comment:
Maybe we should warn about syncing when renaming as well?

Reply:
In this PR renaming is still only local to the buffer.

OCaml-LSP tries to do smart things with syntactic sugar like punning, that require more work to do properly in the case of project-wide renaming (we need to load and parse every involved source file). I don't have a clean patch for that yet.